src/codemagic/models/private_key.py

```diff
@@ -84,8 +70,25 @@ def from_openssh_key(cls, buffer: AnyStr, password: Optional[AnyStr] = None) ->
 
     @classmethod
     def from_pem(cls, pem_key: AnyStr, password: Optional[AnyStr] = None) -> PrivateKey:
-        pkey = cls._get_pkey(pem_key, cls._bytes(password) if password else b'')
-        return PrivateKey(pkey.to_cryptography_key())
+        _password: Optional[bytes] = cls._bytes(password) if password is not None else password
+
+        try:
+            cryptography_private_key = serialization.load_pem_private_key(cls._bytes(pem_key), _password)
+        except ValueError as ve:
+            if 'bad decrypt' in str(ve).lower():
+                log.get_file_logger(cls).exception('Failed to initialize private key: Invalid password')
+                raise ValueError('Invalid private key passphrase') from ve
+            else:
+                log.get_file_logger(cls).exception('Failed to initialize private key: Invalid PEM contents')
+                raise ValueError('Invalid private key PEM content') from ve
+        except TypeError as te:
```

Comment:
Isn't this case covered by `ValueError`? Just wanted to clarify if both errors are possible for invalid passphrase.

Reply:
There is a difference in how `cryptograhy` and `OpenSSL` handle keys that are non-encrypted or signed with empty passwords. `crypto.load_privatekey` from `OpenSSL` treats those two the same, by raising `crypto.Error` in both cases. However, `serialization.load_pem_private_key` from `cryptograhpy` will raise `ValueError` in one case and `TypeError` in other, distinguishing those two cases.